### PR TITLE
TS-Assignment 

### PR DESCRIPTION
**Pull request by sah-is-classic**
please do look into MyObvervations.txt... Thanks :)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,9 +10,11 @@
...
     "@reduxjs/toolkit": "^1.9.1",
+    "antd": "^5.1.7",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
-    "react-redux": "^8.0.5"
+    "react-redux": "^8.0.5",
+    "react-router-dom": "^6.8.0"
   },
   "devDependencies": {
     "@types/react": "^18.0.26",
```
Comment (atmeshwar-classic) [inline package.json:20]: We already have a yarn.lock file. We have no need for this.
Also, it is a bad practice to keep 2 lockfiles.

So, please delete this one and install dependencies using yarn.

File: package-lock.json [hunk 2]
```diff
@@ -0,0 +1,1669 @@
+{
```
Comment (atmeshwar-classic) [inline package-lock.json:1]: We already have a yarn.lock file. We have no need for this.
Also, it is a bad practice to keep 2 lockfiles.

So, please delete this one and install dependencies using yarn.

File: src/App.tsx [hunk 3]
```diff
@@ -1,34 +1,21 @@
...
-      <p className="read-the-docs">
-        Click on the Vite and React logos to learn more
-      </p>
+      <BrowserRouter>
+        <Navbar />
+        <Routes>
+          <Route path="/" element={<BooksPage />} />
+          <Route path="/Cart" element={<CartPage />} />
+        </Routes>
+      </BrowserRouter>
```
Comment (atmeshwar-classic) [inline src/App.tsx:16]: We should always write our routes in a separate Routes file and then import them here.
It alright for now but please keep it in mind for future.

We should always maintain our folder structure regardless of the size of the app.

File: src/components/Navbar/Navbar.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
+// import React, { useState } from "react";
+// import { Drawer } from "antd";
+import { useDispatch, useSelector } from "react-redux";
+import { RootState } from "../../store/store";
+import { changeToCartPage, changeToBookPage } from "../../domain/app/appSlice";
```
Comment (atmeshwar-classic) [inline src/components/Navbar/Navbar.tsx:5]: I don't think so we need these actions anymore as we are using routes.

File: src/components/Navbar/Navbar.tsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+import { useNavigate } from "react-router-dom";
+import "./style.css";
+
+const Navbar = () => {
+  // const [open, setOpen] = useState(false);
+  const cartdata = useSelector((state: RootState) => state.cart.cart);
```
Comment (atmeshwar-classic) [inline src/components/Navbar/Navbar.tsx:11]: Do not write the selectors here, instead move it to its respective slice.
Inside cartslice.ts, create a cartSelector just like in booksSlice.

File: src/domain/app/appSlice.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import { AppState } from "../../types";
```
Comment (atmeshwar-classic) [inline src/domain/app/appSlice.ts:1]: Do we really need this slice?

File: src/domain/books/BooksPage.tsx [hunk 7]
```diff
@@ -1,7 +1,45 @@
...
+const BooksPage = () => {
+  const dispatch: AppDispatch = useDispatch();
+  const Bookdata = useSelector((state: RootState) => state.books.books);
+  const [bookData, setBookData] = useState<Book[]>(Bookdata);
+  useEffect(() => {
+    dispatch(getBooks());
+  }, []);
+  useEffect(() => {
+    setBookData(Bookdata);
+  }, [Bookdata]);
```
Comment (atmeshwar-classic) [inline src/domain/books/BooksPage.tsx:19]: If we are storing a state in redux then it doesn't make any sense to copy it to local component.
Also, use the create the selectors inside the slice files and use them here.

`const books = useSelector(booksSelector)`

File: src/domain/cart/CartPage.tsx [hunk 8]
```diff
@@ -1,5 +1,39 @@
+import { useSelector } from "react-redux";
+import "./style.css";
+import { RootState } from "../../store/store";
+import CartCard from "../../components/cartCard/CartCard";
+import { CartItem } from "./types";
+
 export const CartPage = (): JSX.Element => {
+  const cartdata = useSelector((state: RootState) => state.cart.cart);
```
Comment (atmeshwar-classic) [inline src/domain/cart/CartPage.tsx:8]: Like above, create selectors inside slice files and use them here.

**Review by atmeshwar-classic — CHANGES_REQUESTED** [review]
Hi @sah-is-classic I have left the comments, please check them.

File: package-lock.json (hunk 2 above)
Comment (sah-is-classic) [reply package-lock.json:1]: sure i will

File: src/App.tsx (hunk 3 above)
Comment (sah-is-classic) [reply src/App.tsx:16]: I will keep that  in mind for sure!

File: src/components/Navbar/Navbar.tsx (hunk 4 above)
Comment (sah-is-classic) [reply src/components/Navbar/Navbar.tsx:5]: yeah should have commented code wiz not important my bad...

File: src/domain/books/BooksPage.tsx (hunk 7 above)
Comment (sah-is-classic) [reply src/domain/books/BooksPage.tsx:19]: yes they is no need to do that locally... I will change that

File: src/domain/app/appSlice.ts (hunk 6 above)
Comment (sah-is-classic) [reply src/domain/app/appSlice.ts:1]: not now but just keeping it maybe for future, I will comment it ....

File: src/components/Navbar/Navbar.tsx (hunk 5 above)
Comment (sah-is-classic) [reply src/components/Navbar/Navbar.tsx:11]: sure

File: src/domain/cart/CartPage.tsx (hunk 8 above)
Comment (sah-is-classic) [reply src/domain/cart/CartPage.tsx:8]: sure...

**Comment by atmeshwar-classic** [comment]
I am closing this PR as the assessment is finished.
Good job @sah-is-classic . Just keep in mind the pointers I had shared and you would grow into a better dev.

**Comment by sah-is-classic** [comment]
sure Sir I will keep your comments in mind, Thanks a lot :)

**Pull request closed without merging**